uvicorn/protocols/utils.py

```diff
@@ -14,8 +15,12 @@ def get_remote_addr(transport):
         else:
             family = socket_info.family
 
-        if family in (socket.AF_INET, socket.AF_INET6):
-            return (str(info[0]), int(info[1]))
+        if not sys.platform.startswith("win"):
+            if family in (socket.AF_INET, socket.AF_INET6, socket.AF_UNIX):
```

Comment:
How do you feel about extracting this tuple from function and using attr error instead of os snuffing for feature detection?

```python
try:
   SUPPORTED_AF = (socket.AF_INET, socket.AF_INET6, socket.AF_UNIX)
except AttributeError:
   SUPPORTED_AF = (socket.AF_INET, socket.AF_INET6)
```

Reply:
Agreed. I'd use something like @rafalp's suggestion here, and pull it into a top level constant.

Eg.

```python
if hasattr(socket, 'AF_UNIX'):
    SUPPORTED_SOCKET_FAMILIES = (socket.AF_INET, socket.AF_INET6, socket.AF_UNIX)
else:
    SUPPORTED_SOCKET_FAMILIES = (socket.AF_INET, socket.AF_INET6)
```

Reply:
way better indeed